### DISCUSSION TITLE
Request for the FID Evaluation Code in MAISI

### DISCUSSION THREAD

**Discussion by mar-cry** [Project-MONAI/MONAI#8243] [Q&A] [answered]
I am currently developing a CT image generation algorithm and would like to compare its performance with the results reported in your work, MAISI. However, we have observed a significant discrepancy between the FID values calculated using our code and those reported in your paper.

To ensure fairness in the comparison, I would like to kindly request the code or implementation details used in MAISI for calculating the FID between generated and real images. This would greatly help in maintaining consistency and accuracy in our evaluations.

**Comment by aylward** [comment]
@guopengf ?

**Comment by guopengf** [comment]
We are using the [FIDMetric](https://github.com/Project-MONAI/MONAI/blob/e6cae1ced23f32feb2f42943d15b7dd49481c794/monai/metrics/fid.py#L23) in MONAI and the feature extractor is defined as follows: 
`feature_network = torch.hub.load(
            "Warvito/radimagenet-models",
            model="radimagenet_resnet50",
            verbose=True,
            trust_repo=True,
        )`.

@dongyang0122 Shall we release the FID evaluation script in the [MAISI MONAI Tutorial Repo](https://github.com/Project-MONAI/tutorials/tree/main/generation/maisi)?

**Reply by mar-cry** [reply]
Thank you very much for providing the testing method. I will first try to calculate the FID using the method you provided. Additionally, I would greatly appreciate it if you could share the complete FID testing script.

**Reply by danielemolino** [reply]
@mar-cry were you able to compute the FID score?

**Comment by Can-Zhao** [comment]
@dongyang0122 This was the discussion we mentioned last week. Thank you!

**Comment by pshavela** [comment]
Hello, 
is there any update on this? 
Is a global spatial average applied on the extracted features for flattening?
Also, would it be possible to share the SSIM code?

**Reply by dongyang0122** [reply]
You may refer this part for FID computation now.
https://github.com/Project-MONAI/tutorials/tree/main/generation/maisi#4-fid-score-computation

**Comment by dongyang0122** [comment]
@mar-cry Thanks for your interest. We will be adding the FID computation script to this tutorial in the following days.

**Reply by danielemolino** [reply]
@dongyang0122 Any news?

**Reply by dongyang0122** [reply]
You may refer this part for FID computation now.
https://github.com/Project-MONAI/tutorials/tree/main/generation/maisi#4-fid-score-computation

**Comment by dongyang0122** [comment, marked as answer]
> I am currently developing a CT image generation algorithm and would like to compare its performance with the results reported in your work, MAISI. However, we have observed a significant discrepancy between the FID values calculated using our code and those reported in your paper.
> 
> To ensure fairness in the comparison, I would like to kindly request the code or implementation details used in MAISI for calculating the FID between generated and real images. This would greatly help in maintaining consistency and accuracy in our evaluations.

You may refer this part for FID computation now.
https://github.com/Project-MONAI/tutorials/tree/main/generation/maisi#4-fid-score-computation
